### PR TITLE
[nexus] Silo IP pools schema change

### PR DESCRIPTION
**Pull request by david-crespo**
This is quite small, but I'm pretty confident it's a sufficient basis for the rest of the work in #3926, so we might as well review and merge already.

A couple of minor pain points I ran into while doing this:

* Needed guidance on version number (3.0.0 it is) — it will be more obvious when there are more, but I added a line to the readme anyway
* Diff output for [`dbinit_equals_sum_of_all_up`](https://github.com/oxidecomputer/omicron/blob/fb16870de8c0dba92d0868a984dd715749141b73/nexus/tests/integration_tests/schema.rs#L601) is terrible — thousands of lines with only a few relevant ones. It turns out the column order matters, so if you add columns to a table in a migration, you have to add them to the _end_ of the `create table` in `dbinit.sql`.

### CODE REVIEW

**Review by smklein — APPROVED** [review]
The mechanics of updating the DB look correct to me 👍

File: schema/crdb/dbinit.sql [hunk 1]
```diff
@@ -1480,12 +1480,20 @@ CREATE TABLE IF NOT EXISTS omicron.public.ip_pool (
...
 
     /* The collection's child-resource generation number */
-    rcgen INT8 NOT NULL
+    rcgen INT8 NOT NULL,
+
+    /*
+     * Fields representating association with a silo or project. silo_id must be
+     * non-null if project_id is non-null. When project_id is non-null, silo_id
+     * will (naturally) be the ID of the project's silo. Both must be null if
+     * internal is true, i.e., internal IP pools must be fleet-level pools.
```
[the remark this answers is not in the document]
Comment (david-crespo) [reply schema/crdb/dbinit.sql:1493]: Definitely — I didn't know you could do that.

**Review by ahl — APPROVED** [review]
looks good to me modulo @smklein's suggestion

File: schema/crdb/dbinit.sql (hunk 1 above)
[the remark this answers is not in the document]
Comment (luqmana) [reply schema/crdb/dbinit.sql:1493]: Sorry didn't see this yesterday, was there a specific reason internal pools couldn't be silo or project scoped? We already have a separate silo for control plane services and create project, vpc, etc in it during RSS. We could totally just scope the internal ip pool to that silo/project. And in fact, with that we could even just drop the internal flag on ip pools.

**Pull request merged**